sources/src/ChatGPT.tsx

```diff
@@ -249,6 +250,11 @@ function ChatGPT(props: ChatGPTProps) {
   const maxMessageIndex = messages.length - 1;
   const srcDoc = messages.length ? createSrcDoc('...', theme) : '';
 
+  const handleInputChange = (value: string) => {
+    setPrompt(value);
+    hasConversationRef.current = value.length > 0;
```

Comment:
Having this here will set hasConversation to false even when there is a conversation if someone would type stuff and delete it leaving the input blank (after having sent previous messages). I think `hasConversation` should depend solely on `messages.length > initialMessages.length`.
```suggestion
```

Reply:
@rart I updated to use both the `messages` length and the `promp` length:
```
hasConversationRef.current = messages.length > (initialMessages?.length ?? 0) || prompt.length > 0;
```

If there is no messages but user input something in the prompt, I think we should consider this as has converstation.